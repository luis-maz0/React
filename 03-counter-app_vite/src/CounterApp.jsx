import { useState } from "react";
import PropTypes from "prop-types"; 

export const CounterApp = ({ value})=>{
  
    const [counter, setCounter] = useState( value ); 

    const increaseCounter = () => {
        setCounter(counter + 1)
    }
    const decreaseCounter = () =>{
        setCounter(counter - 1)
    }
    const resetCounter = ()=>{
        setCounter(value)
    }

    return(
        <>
            <h1>CounterApp</h1>
            <h2> { counter }</h2>

            <button onClick={ increaseCounter }>+1</button>
            <button onClick={ decreaseCounter }>-1</button>
            <button aria-label="btn-reset" onClick={ resetCounter  }>Reset</button>
        </>
    )
}

CounterApp.propTypes = {
    value : PropTypes.number.isRequired,
}
