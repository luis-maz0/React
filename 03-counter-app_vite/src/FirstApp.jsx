import PropTypes from 'prop-types';

const getSaludo = ()=> "Hola mundo"; 
const array = [1,2,3,4]; 
const objeto = { 
    titulo: "First App"
}

export const FirstApp = ( {saludo, edad, nombre } )=> {
    return(
        <> 
            {/*
            <h2>Titulo objeto: { props.title }</h2>
             <p>Mensaje retorno de funci√≥n: { getSaludo() } </p>
            <p>Contenido de array: { array } </p>
            <p> {true } </p>
            <code>Visualizar objeto:  {JSON.stringify(objeto)} </code>
             */}
             <h1 data-testid="test-saludo">{saludo}</h1>
             <p data-testid="test-p">{nombre}, {edad}</p>
        </>
)}

FirstApp.propTypes = {
    saludo: PropTypes.string.isRequired,
    edad: PropTypes.number,
    nombre: PropTypes.string
};

FirstApp.defaultProps = {
    nombre: "NO HAY NOMBRE",
    edad: 0
}