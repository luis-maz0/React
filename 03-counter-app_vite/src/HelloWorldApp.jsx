//Funcional component
//Componentes basados en funciones
const HelloWorldApp =  () => <h1> Hola mundo</h1> 

export{
    HelloWorldApp
}