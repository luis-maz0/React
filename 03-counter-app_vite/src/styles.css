*{
    margin: 0;
    padding: 0;
    box-sizing: border-box;
}
body{
    background-color: black;
    color: white;
    font-size: 1.3rem;
    padding: 70px;
}
button{
    padding: .30rem .5rem;
    font-size: 1.3rem;
    padding: .5rem;
    margin-right: 1rem;
}