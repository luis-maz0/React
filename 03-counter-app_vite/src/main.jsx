import React from "react";
import ReactDOM from "react-dom/client";
import { FirstApp } from "./FirstApp";
//import { HelloWorldApp } from "./HelloWorldApp";
import "./styles.css";
import { CounterApp } from "./CounterApp";

ReactDOM.createRoot( document.querySelector("#root")).render( 
    <React.StrictMode>
        {/* <FirstApp saludo="Titulo"/> */}
        <CounterApp value={ 100 }/>
    </React.StrictMode>
)