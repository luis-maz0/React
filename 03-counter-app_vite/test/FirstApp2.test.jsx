import { render, screen} from "@testing-library/react"
import { FirstApp } from "../src/FirstApp"

describe('Pruebas optimizadas FirstApp ', () => {
    //Utilizamos constantes globales para todos los test
    const textoSaludo = "HOLA",
          textoNombre = "luis",
          textoEdad = 25; 

    test('Hacer match con el snapshot', () => {
        const {container} = render( <FirstApp
            saludo={ textoSaludo }
            nombre={ textoNombre }
            edad={ textoEdad }
        />)
        expect( container ).toMatchSnapshot()
      });
    
    test("Debe existir el mensaje del saludo",()=>{
       render( <FirstApp
            saludo={ textoSaludo }
            nombre={ textoNombre }
            edad={ textoEdad }
        />)
        //Mostrar el componente entero renderizado
        screen.debug() 

        //Verificar existencia del saludo
        expect( screen.getByText(textoSaludo)).toBeTruthy()
    })
    test('Debe mostrar el contenido del h1', () => {
        render( <FirstApp
            saludo={ textoSaludo }
            nombre={ textoNombre }
            edad={ textoEdad }
        />)
        expect( screen.getByRole("heading", {level:1}).innerHTML).toContain(textoSaludo)
    })
  })