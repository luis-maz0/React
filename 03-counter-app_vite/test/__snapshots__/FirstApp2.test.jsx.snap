// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas optimizadas FirstApp  Hacer match con el snapshot 1`] = `
<div>
  <h1
    data-testid="test-saludo"
  >
    HOLA
  </h1>
  <p
    data-testid="test-p"
  >
    luis
    , 
    25
  </p>
</div>
`;
